### ISSUE TITLE
[BUG] Vai lists on resources causes log spam

### ISSUE THREAD

**Issue by MbolotSuse** [rancher/rancher#46030]
**Rancher Server Setup**
- Rancher version: v2.9-head
- Installation option (Docker install/Helm Chart): N/A - dev build, but should be on setups
   - If Helm Chart, Kubernetes Cluster and version (RKE1, RKE2, k3s, EKS, etc):
- Proxy/Cert Details:

**Information about the Cluster**
- Kubernetes version:
- Cluster Type (Local/Downstream): Local, k3d, v1.30.1+k3s1

**User Information**
- What is the role of the user logged in? Admin

**Describe the bug**
After requesting an un-watchable resources (e.g. `nodemetrics.metrics.k8s.io`), Rancher gets repeated informer errors in the logs, like:
```
E0705 16:02:31.280985   90705 reflector.go:150] pkg/mod/k8s.io/client-go@v0.30.1/tools/cache/reflector.go:232: Failed to watch metrics.k8s.io/v1beta1, Kind=NodeMetrics: the server does not allow this method on the requested resource
```

**To Reproduce**
1. Start rancher/rancher:v2.9-head
2. Enable the `ui-sql-cache` feature flag.
3. Go to `https://$RANCHER/v1/metrics.k8s.io.nodemetrics` in your browser (or access the endpoint through another mechanism).

**Result**

Repeated log messages like the below begin to appear:

```
E0705 16:02:31.280985   90705 reflector.go:150] pkg/mod/k8s.io/client-go@v0.30.1/tools/cache/reflector.go:232: Failed to watch metrics.k8s.io/v1beta1, Kind=NodeMetrics: the server does not allow this method on the requested resource
```

**Expected Result**
No repeated log messages.

**Screenshots**
N/A.

**Additional context**
This problems was observed for a few resources, including:
- `subjectaccessreview.authorization.k8s.io`
- `podmetrics.metrics.k8s.io`

The core of the issue seems to be that vai starts an informer for each type after a request is made for that type, even if the type isn't watchable (like the nodemetrics type, which you can get but can't watch).

**Comment by moio** [comment]
Research notes:

- `subjectaccessreview` does not allow `list` either, so it is out of scope of Vai at this point
- according to the following commands, listable not watchable resources are:

```sh
kubectl api-resources --verbs=list --no-headers -o wide | awk '{print $1 " " $2}' > listable
kubectl api-resources --verbs=watch --no-headers -o wide | awk '{print $1 " " $2}' > watchable
grep -Fxv -f watchable listable
```

In a freshly deployed k3s setup, output is:
```
componentstatuses cs
nodes metrics.k8s.io/v1beta1
pods metrics.k8s.io/v1beta1
```

 - componentstatuses is deprecated in Kubernetes v1.19+ and not used in the UI (dashboard)
 - node and pod metrics are used in the UI (dashboard)
 - it is always possible users set up permissions that way

**Comment by moio** [comment]
Once steve#284 is merged this can be merged, as the rest of this issue is covered by https://github.com/rancher/rancher/issues/46768
